### ISSUE TITLE
[Android] browser sample - keyboard app lockup on minimize attempt

### ISSUE THREAD

**Issue by rgdev78** [gameplay3d/GamePlay#1626]
So far I'm loving GamePlay3D - next very 3.0.0.  I got the browser sample on my Android 4.2.2 tablet just fine but noticed several issues.

The most notably is that the demo for ui input when I click to input with the keyboard it works fine

But

When I click the minimize arrow to minimize the keyboard then it highlights and the keyboard becomes unresponsive then if I goto the main Android desktop then go back to the browser sample while its running then I get a msg from Android asking if I want to close or wait.

I click on close and it goes back to the main browser sample menu.

That tells me that something is making the android keyboard "app" itself lockup instead of minimizing it.

**Comment by zet23t** [comment]
I can confirm that problem with my Samsung Galaxy Note III neo in combination with the swipe keyboard I've installed. The app becomes unresponsive and the system is sluggish. Changing back to the home menu takes a while and most notably, the virtual keyboard remains active (which is very weird) until the app is terminated forcefully. I have not yet investigated the issue further (logcat for example) but simply searched for open issues... in case I have some more free time, I'll look into the issue.

**Comment by reven86** [comment]
I found a reason for this behavior. There is a __displayKeyboard flag in PlatformAndroid.cpp:
```
static bool __displayKeyboard = false;
```
Every frame Gameplay tries to show/hide keyboard in the message loop (Platform::enterMessagePump):
```
        gameplay::displayKeyboard(__state, __displayKeyboard);
```
But when the user hides keyboad, there is no way Gameplay know about that and the __displayKeyboad flag remains true. Next frame the Gameplay tries to show keyboard again since __displayKeyboard wasn't changed. One possible solution would be to remove __displayKeyboard flag and show/hide keyboard right in the Platform::displayKeyboard as it done for iOS.

**Comment by reven86** [comment]
Ok, I've figured it out. My previous pull request actually fix another issue for old android version, so I've closed it. The actual issue with keyboard lockup is related to bug in andoid ndk for Android 4.1 and 4.2 (see http://stackoverflow.com/questions/15913080/crash-when-closing-soft-keyboard-while-using-native-activity), it's fixed for later versions of Android. I have a workaround for this issue and will open pull request again once I've tested it on several devices.
